### PR TITLE
Chore: Fix/remove tools (fixes #471)

### LINKED ISSUE #471: Fix broken scripts in tools/ and convert to esm
As per discussion in  https://github.com/eslint/espree/pull/469#issuecomment-792229606 there are several files in the `tools/` directory that are not functioning, and are also declared as commonjs modules.

### PR DESCRIPTION
**Pull request by nzakas**
1. I removed a bunch of scripts in the `tools/` directory that were left over from the initial Esprima migration and were no longer used.
2. For tools that were heavily reliant on `require()`, I opted to just rename the file to `.cjs`.

### CODE REVIEW

**Review by btmills — APPROVED** [review]

**Review by mdjermanovic — CHANGES_REQUESTED** [review]
We should also update both `create-test.js` and `update-tests.js` to produce ESM content in `*.result.js` files, because the tests will import those files and all `*.js` files are now ESM. In particular, we should replace `module.exports =` with `export default` [here](https://github.com/eslint/espree/blob/5dd3dee83616838451adad7183dd3dd8499a7335/tools/create-test.js#L97) and [here](https://github.com/eslint/espree/blob/5dd3dee83616838451adad7183dd3dd8499a7335/tools/update-tests.js#L54).

File: tools/update-tests.cjs [hunk 1]
```diff
@@ -15,6 +12,8 @@
 // Requirements
 //------------------------------------------------------------------------------
 
+"use strict";
+
 var shelljs = require("shelljs"),
     espree = require("../espree"),
     tester = require("../tests/lib/tester"),
```
[the remark this answers is not in the document]
Comment (nzakas) [reply tools/update-tests.cjs:19]: Doh! Unsure how I missed this. Thanks.

**Comment by nzakas** [comment]
Should be set now. Also found that the scripts didn't work with the bigint tests, so fixed that too.

**Review by mdjermanovic — APPROVED** [review]
LGTM!

**Comment by mdjermanovic** [comment]
There's a merge conflict in `package.json`.

**Comment by nzakas** [comment]
Conflict fixed.

**Pull request merged**